lib/sql-manager/index.js

```diff
@@ -9,60 +9,55 @@ try {
      * @param   {ConnectionData}         connection   The connection data for the SQL DB.
      * @param   {DatabaseManagerOptions} [options={}] The DatabaseManagerOptions.
      */
-    constructor (connection, options) {
-      super(options)
-      /**
-       * @protected
-       * @type      {QueryBuilder}
-       */
-      this._qb = new QueryBuilder({
-        ...connection,
-        pool: {
-          min: 0,
-          ...connection.pool
+    constructor (
+      {
+        client,
+        connectionInfo: {
+          pool,
+          ...connection
         }
-      })
+      },
+      options
+    ) {
+      super(options)
+      this._qb = knex({ client, connection, pool: { min: 0, ...pool } })
     }
 
     async add (type, data) {
-      const [ res ] = await this._qb.insert({ table: type, data })
+      const [ res ] = await this._qb(type)
+        .insert(data, '*')
 
-      return {
-        ...res
-      }
+      return res
     }
 
     async delete (object) {
-      await this._qb.delete({
-        table: object.type,
-        where: { id: object.id }
-      })
+      await this._qb(object.type)
+        .where({ id: object.id })
+        .del()
     }
 
     async update (object) {
-      const [ res ] = await this._qb.update({
-        table: object.type,
-        data: object._data,
-        where: { id: object.id }
-      })
-
-      return {
-        ...res
-      }
+      const [ res ] = await this._qb(object.type, '*')
+        .where({ id: object.id })
+        .update(object._data)
+
+      return res
     }
 
     async get (query) {
-      return this._qb.get({
-        table: query.type,
-        where: { [query.getKey]: query.getValue }
-      })
+      const rows = await this._qb(query.type)
+        .select('*')
+        .where({ [query.getKey]: query.getValue })
+
+      return rows[0]
     }
 
     async find (query) {
-      return this._qb.select({
-        table: query.type,
-        where: (builder) => this._whereBuilder(query, builder)
-      })
+      const rows = await this._qb(query.type)
+        .select('*')
+        .where((builder) => this._whereBuilder(query, builder))
+
+      return rows[0] ? rows : []
     }
 
     _whereBuilder (query, builder) {
```

Comment:
This is very mushy code gravy. Beautify it and clean it up a bit.

Reply:
explain

Reply:
It's extremely difficult to understand anything going on in there. It's very squished together and hard to follow